packages/vscode-extension/src/commonlib/azureLogin.ts

```diff
@@ -447,10 +483,97 @@ export class AzureAccountManager extends login implements AzureAccountProvider {
         throw result.error;
       } else {
         const subId = result.value.result as string;
+        subscriptionList.filter(async (item) => {
+          if (item.subscriptionId === subId) {
+            await this.saveSubscription({
+              subscriptionId: item.subscriptionId,
+              subscriptionName: item.subscriptionName,
+              tenantId: item.tenantId,
+            });
+          }
+        });
         await this.setSubscription(subId);
       }
     }
   }
+
+  async saveSubscription(subscriptionInfo: SubscriptionInfo): Promise<void> {
+    const subscriptionFilePath = await this.getSubscriptionInfoPath();
+    if (!subscriptionFilePath) {
+      return;
+    } else {
+      await fs.writeFile(subscriptionFilePath, JSON.stringify(subscriptionInfo, null, 4));
+    }
+  }
+
+  async readSubscription(): Promise<SubscriptionInfo | undefined> {
+    const subscriptionFilePath = await this.getSubscriptionInfoPath();
+    if (!subscriptionFilePath || !fs.existsSync(subscriptionFilePath)) {
+      const solutionSubscriptionInfo = await this.getSubscriptionInfoFromEnv();
+      if (solutionSubscriptionInfo) {
+        await this.saveSubscription(solutionSubscriptionInfo);
+        return solutionSubscriptionInfo;
+      }
+      return undefined;
+    } else {
+      const content = (await fs.readFile(subscriptionFilePath)).toString();
+      const subcriptionJson = JSON.parse(content);
+      return {
+        subscriptionId: subcriptionJson[subscriptionIdString],
+        tenantId: subcriptionJson[tenantIdString],
+        subscriptionName: subcriptionJson[subscriptionNameString],
+      };
+    }
+  }
+
+  async getSubscriptionInfoPath(): Promise<string | undefined> {
+    if (vscode.workspace.workspaceFolders) {
+      const workspaceFolder: vscode.WorkspaceFolder = vscode.workspace.workspaceFolders[0];
+      const workspacePath: string = workspaceFolder.uri.fsPath;
+      if (!(await commonUtils.isFxProject(workspacePath))) {
+        return undefined;
+      }
+      const configRoot = await commonUtils.getProjectRoot(
+        workspaceFolder.uri.fsPath,
+        `.${ConfigFolderName}`
+      );
+      const subscriptionFile = path.join(configRoot!, subscriptionInfoFile);
+      return subscriptionFile;
+    } else {
+      return undefined;
+    }
+  }
+
+  async getSubscriptionInfoFromEnv(): Promise<SubscriptionInfo | undefined> {
+    if (vscode.workspace.workspaceFolders) {
+      const workspaceFolder: vscode.WorkspaceFolder = vscode.workspace.workspaceFolders[0];
+      const workspacePath: string = workspaceFolder.uri.fsPath;
+      if (!(await commonUtils.isFxProject(workspacePath))) {
```

Comment:
Why `commonUtils` is from *debug/* folder... is there a more common utility?

Reply:
I find only env checker and debug have this feature.